Sources/AsyncQueue/AsyncQueue.swift

```diff
@@ -20,4 +20,70 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-final class AsyncQueue {}
+/// A queue that enables ordered sending of events from synchronous to asynchronous code
+public final class AsyncQueue: Sendable {
+
+    // MARK: Initialization
+
+    /// Instantiates an asynchronous queue.
+    /// - Parameter priority: The priority of the tasks added to the asynchronous queue.
+    public init(priority: TaskPriority = .medium) {
```

Comment:
```suggestion
    public init(priority: TaskPriority? = nil) {
```
Let's follow the Swift library convention for using `TaskPriority` where `nil` means we don't care and let the system figure it out. May need to tweak some of the other properties as well.

Reply:
Good call!